sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```
@@ -93,6 +93,28 @@ object IntervalUtils {
 
   private val yearMonthPattern = "^([+|-])?(\\d+)-(\\d+)$".r
 
+  private val yearMonthFuzzyPattern = "[^-|+]*?([+|-]?[\\d]+-[\\d]+).*".r
+
+  def safeFromYearMonthString(input: UTF8String): Option[Int] = {
```

Why not just call `fromYearMonthString`?

---

> Why not just call `fromYearMonthString`?

Null safe? like `safeStringToInterval` we can avoid NPE

---

It's a new feature, we can use the ANSI behavior directly.

---

> It's a new feature, we can use the ANSI behavior directly.

So just let it throw exception? Update later

---

How about current?